From shemminger@linux-foundation.org Mon Feb 26 17:36:20 2007
From: Stephen Hemminger <shemminger@linux-foundation.org>
Date: Mon, 26 Feb 2007 17:18:38 -0800
Subject: sky2: dont flush good pause frames
To: Greg KH <greg@kroah.com>
Cc: stable@kernel.org
Message-ID: <20070227012021.809906985@linux-foundation.org>
Content-Disposition: inline; filename=sky2-pause-flush.patch

From: Stephen Hemminger <shemminger@linux-foundation.org>

Don't mark pause frames as errors. This caused a bug that would
lockup transmitter, as well as all the other ports on a switch.

Signed-off-by: Stephen Hemminger <shemminger@linux-foundation.org>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/net/sky2.h |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- linux-2.6.19.5.orig/drivers/net/sky2.h
+++ linux-2.6.19.5/drivers/net/sky2.h
@@ -1576,7 +1576,7 @@ enum {
 
 	GMR_FS_ANY_ERR	= GMR_FS_RX_FF_OV | GMR_FS_CRC_ERR |
 			  GMR_FS_FRAGMENT | GMR_FS_LONG_ERR |
-		  	  GMR_FS_MII_ERR | GMR_FS_GOOD_FC | GMR_FS_BAD_FC |
+		  	  GMR_FS_MII_ERR | GMR_FS_BAD_FC |
 			  GMR_FS_UN_SIZE | GMR_FS_JABBER,
 };
 
